src/LayerPlan.cpp

```diff
@@ -459,7 +459,15 @@ GCodePath& LayerPlan::addTravel(Point p, bool force_comb_retract)
             }
         }
     }
-    
+
+    // CURA-6675:
+    // Retraction Minimal Travel Distance should work for all travel moves. If the travel move is larger than the
+    // Retraction Minimal Travel Distance, retraction should be disabled.
+    if (!is_first_travel_of_layer && last_planned_position && shorterThen(*last_planned_position - p, retraction_config.retraction_min_travel_distance))
```

Comment:
This is duplicating the test that follows with a minor change. If you are going to do this you should refactor the test below and avoid the duplication.